src/library/cratefeature.cpp

```diff
@@ -128,17 +143,71 @@ int CrateFeature::crateIdFromIndex(QModelIndex index) {
 
     QString dataPath = item->dataPath().toString();
     bool ok = false;
-    int playlistId = dataPath.toInt(&ok);
+    int crateId = dataPath.toInt(&ok);
     if (!ok) {
         return -1;
     }
-    return playlistId;
+    return crateId;
+}
+
+QModelIndex CrateFeature::indexFromCrateId(int id) {
+
+    QStack<QModelIndex> cratesToProcess;
+    QModelIndex crate;
+    cratesToProcess.push(QModelIndex());
+    while (! cratesToProcess.empty()) {
+        crate = cratesToProcess.pop();
+
+        if (id == crateIdFromIndex(crate)) {
+            qDebug() << "Found index of crate found id: " << id;
+            return crate;
+        }
+
+        for(int row = 0; row < m_childModel.rowCount(crate); row++) {
+            cratesToProcess.push(m_childModel.index(row, 0, crate));
+        }
+    }
+    qDebug() << "Failed search for index of crate id: " << id;
+    return QModelIndex();
+}
+
+// updates hasChildren function in childModel
+void CrateFeature::refreshModelHasChildren() { 
+    //put the set of folders to variable to be lambda-captureable
+    QSet<int> folderSet = m_crateDao.getFolderSet();
+    m_childModel.setHasChildrenFunction( 
+        [folderSet] (QVariant crateId) {
+             return folderSet.contains(crateId.toInt());
+        }
```

Comment:
Year, lamdas a pretty cool, but It looks like we do not need them in this context.
Cant we just pass the folderSet and put the Lamda content directly to m_childModel? This would streamline the code.